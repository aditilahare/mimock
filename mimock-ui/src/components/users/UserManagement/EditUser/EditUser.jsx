import React from 'react';
import { IconHeader } from 'src/styles/components';
import { FaEdit } from 'react-icons/fa';
import { EditUserWrapper, EditUserHeader } from './EditUser.style';
import UserDetails from './UserDetails';

export default function EditUser() {
	return (
		<EditUserWrapper data-testid='edit-user'>
			<EditUserHeader>
				<IconHeader
					icon={<FaEdit />}
					title='Edit User'
					dataTestId='edit-user-header'
					enableBackNavigation={true}
				/>
			</EditUserHeader>
			<UserDetails />
		</EditUserWrapper>
	);
}
