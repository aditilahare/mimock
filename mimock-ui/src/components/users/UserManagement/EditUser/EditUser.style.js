import tw from 'tailwind-styled-components';

export const EditUserWrapper = tw.div`
    w-11/12
    h-full
    mx-auto
    my-10
`;

export const EditUserHeader = tw.div`
    w-full
    py-4
    px-4
    border-b-2
    border-gray-100
`;
