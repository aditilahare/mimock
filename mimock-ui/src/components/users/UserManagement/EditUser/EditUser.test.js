import React from 'react';
import { render, act } from '@testing-library/react';
import EditUser from './EditUser';

jest.mock('react-router-dom');
jest.mock('./UserDetails', () => {
	const UserDetailsMock = (
		<div data-testid='edit-user-details'>User Details</div>
	);

	return () => UserDetailsMock;
});

describe('EditUser', () => {
	it('should render EditUser', async () => {
		let tree;
		await act(async () => {
			tree = await render(<EditUser />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('edit-user')).toBeInTheDocument();
		expect(getByTestId('edit-user-header')).toBeInTheDocument();
		expect(getByTestId('edit-user-details')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
