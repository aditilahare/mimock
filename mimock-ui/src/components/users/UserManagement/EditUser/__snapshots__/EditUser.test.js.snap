// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUser should render EditUser 1`] = `
<div>
  <div
    class="w-11/12 h-full mx-auto my-10"
    data-testid="edit-user"
  >
    <div
      class="w-full py-4 px-4 border-b-2 border-gray-100"
    >
      <div
        class="flex items-center align-middle font-sans text-2xl font-bold text-gray-700"
        data-testid="edit-user-header"
      >
        <button
          aria-label="Go back"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-gsDKAQ eEqRuN css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="go-back-btn"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M401.4 224h-214l83-79.4c11.9-12.5 11.9-32.7 0-45.2s-31.2-12.5-43.2 0L89 233.4c-6 5.8-9 13.7-9 22.4v.4c0 8.7 3 16.6 9 22.4l138.1 134c12 12.5 31.3 12.5 43.2 0 11.9-12.5 11.9-32.7 0-45.2l-83-79.4h214c16.9 0 30.6-14.3 30.6-32 .1-18-13.6-32-30.5-32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
        <div
          class="mx-2 select-none"
        >
          Edit User
        </div>
      </div>
    </div>
    <div
      data-testid="edit-user-details"
    >
      User Details
    </div>
  </div>
</div>
`;
